doc/ECMP/Hierarchical-ECMP.md

```diff
@@ -0,0 +1,60 @@
+Hierarchical ECMP
+-------------------------------------------------------------------------------
+ Title       | SAI support for hierarchical ecmp
+-------------|-----------------------------------------------------------------
+ Authors     | Jai Kumar, Broadcom Inc.
+ Status      | In review
+ Type        | Standards track
+ Created     | 07/08/2022: Initial Draft
+ SAI-Version | 
+-------------------------------------------------------------------------------
+
+Hierarchical ECMP provides two level route resolution for equal cost multipaths. First level of route resolution will happen via level 1 next hop group. Level 1 nexthop group members may point directly to a next hop or may resolve via 2nd level of next hop group. Level 2 next hop group will always point to a next hop.
+

```

Comment:
Please add the usecase where you mentioned that the routes initially point to an empty Nexthop Group and why deciding underlay or overlay is required.

Reply:
Please look at the latest commit doc file